test_upstream_apis.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+#### SETUP ####
+
+# The directory this file is in must contain a file named 'sar_endpoint_config' which includes the following fields:
+# client_with_sar_role=
+# secret_for_client_with_sar_role=""
+
+# client_without_sar_role=
+# secret_for_client_without_sar_role=""
+
+# valid_nomis_id=
+# valid_ndelius_id=
```

Comment:
I don't think this is used anywhere - maybe we could make this a bit more flexible by having:

dummy_id="" # NOMIS/nDelius ID that should return no data in the system
valid_id="" # NOMIS/nDelius ID that should return data in the system
expected_id_type="" # Identifier expected by the system, ie. prn / crn
unexpected_id_type="" # Identifier not expected by the system, ie. crn / prn

Then they can set expected_id_type to 'prn' or 'crn' depending on what the service expects?